internal/mode/static/state/dataplane/types.go

```diff
@@ -34,6 +35,11 @@ type Configuration struct {
 	TLSPassthroughServers []Layer4VirtualServer
 	// Upstreams holds all unique http Upstreams.
 	Upstreams []Upstream
+	// DeploymentContext contains metadata about NGF and the cluster.
+	DeploymentContext DeploymentContext
+	// AuxiliarySecrets contains additional secret data, like certificates/keys/tokens that are not related to
+	// Gateway API resources.
+	AuxiliarySecrets map[graph.PlusSecretFileType][]byte
```

Comment:
I like the name `AuxiliarySecrets` because it is data plane agnostic, but the map key kinda ruins it... Might as well just call it what it is.

Reply:
That's fair. What about adding a FIXME to make more generic?

Reply:
What if I just started by renaming to SecretFileType?